services/src/main/java/org/keycloak/services/clienttype/DefaultClientTypeManager.java

```diff
@@ -109,8 +109,14 @@ public ClientModel augmentClient(ClientModel client) throws ClientTypeException
         if (client.getType() == null) {
             return client;
         } else {
-            ClientType clientType = getClientType(client.getRealm(), client.getType());
-            return new TypeAwareClientModelDelegate(clientType, () -> client);
+            try {
+                ClientType clientType = getClientType(client.getRealm(), client.getType());
+                return new TypeAwareClientModelDelegate(clientType, () -> client);
+            } catch(ClientTypeException cte) {
+                logger.errorf("Could not augment client, %s, due to client type exception: %s",
+                        client, cte);
+                return client;
```

Comment:
```suggestion
                throw cte;
```

I am not sure about returning client in this case. If something is bad with clientType, it is rather better to propagate exception rather than just log it and return client IMO. Returned "unaugmented" client can have side-effects like for instance administrator being able to edit client properties, which he is not supposed to edit etc.

Reply:
Yeah that is likely the safer option. Reverted so it will propagate the exception as well as log as an error message.

This might affect loading of other clients, as well. For instance through the admin UI, the listing of clients will fail to load if this exception is thrown. The admin will have to fix the issue with the client type to proceed.